pkg/services/datasourceproxy/datasourceproxy.go

```diff
@@ -46,7 +46,7 @@ func (p *DatasourceProxyService) ProxyDatasourceRequestWithID(c *models.ReqConte
 			c.JsonApiErr(http.StatusForbidden, "Access denied to datasource", err)
 			return
 		}
-		c.JsonApiErr(http.StatusInternalServerError, "Unable to load datasource meta data", err)
+		c.JsonApiErr(http.StatusNotFound, "Unable to load datasource meta data", err)
```

Comment:
do no think this is correct, you not checking if this is the actual error

Reply:
addressed